packages/block-editor/src/store/actions.js

```diff
@@ -793,15 +795,19 @@ export function __unstableMarkAutomaticChange() {
 }
 
 /**
- * Returns an action object used to enable or disable the navigation mode.
+ * Generators that triggers an action used to enable or disable the navigation mode.
  *
  * @param {string} isNavigationMode Enable/Disable navigation mode.
- *
- * @return {Object} Action object
  */
-export function setNavigationMode( isNavigationMode = true ) {
-	return {
+export function * setNavigationMode( isNavigationMode = true ) {
+	yield {
 		type: 'SET_NAVIGATION_MODE',
 		isNavigationMode,
 	};
+
+	if ( isNavigationMode ) {
```

Comment:
Are you sure that it should be announced every time someone switches modes? When editing individual blocks with the keyboard it might be a common pattern to switch between modes.

Reply:
> For example when I enable VoiceOver (⌘F5 in MacOS, works best in Safari), I see the following message when the browser has focus:
> 
> ![Screenshot 2019-11-26 at 13 53 56](https://user-images.githubusercontent.com/1204802/69635248-463e3280-1054-11ea-9336-9c2a8ff292fd.png)

I see it now. We can give it a try. Someone should battle test it though :)

Reply:
It is my understanding that yes, we do need something like this. We received helpful information from a user in https://github.com/WordPress/gutenberg/issues/17088#issuecomment-557873792, who had also created #18583 which was related to the switch.

Reply:
Right, my concern is mostly about the number of details included in the message. Anyway, let's collect feedback and refine only if necessary.